### DISCUSSION TITLE
terminate called after throwing an instance of 'std::bad_alloc'   what():  std::bad_alloc Aborted

### DISCUSSION THREAD

**Discussion by satya-auti** [systemed/tilemaker#488] [General]
I'm facing an issue while generating mbtiles for extracted small osm.pbf (11mb) from another osm.pbf (2.7gb) file. Extracted using osmium tool. 
I'm getting this error multiple times is there any solution?
terminate called after throwing an instance of 'std::bad_alloc'
  what():  std::bad_alloc
Aborted 

<img width="602" alt="image" src="https://user-images.githubusercontent.com/103890980/230840254-7954096e-f692-464e-ae15-fa0ea815528b.png">

**Comment by systemed** [comment]
Did you use `--strategy=smart` when creating the file using osmium?

**Reply by satya-auti** [reply]
`osmium extract -b 72.9719,17.8178,75.0314,19.3831 pnq_ht.osm.pbf -o pune_ht.pbf` 
I have used above command to extract the data.

**Reply by systemed** [reply]
Ok. You need to use `--strategy=smart` otherwise the .pbf will be badly formed (i.e. relations and ways will contain references to nodes that aren't in the file).

**Reply by satya-auti** [reply]
Right now I'm trying to do using `--strategy=smart` .

**Reply by satya-auti** [reply]
I have generated pbf file using `osmium extract -b 78.2109,17.1278,78.7402,17.5926 planet_1.osm.pbf -o hyderabad_planet_v0_6.osm.pbf --strategy=smart` this command, But I'm still facing the same issue file generating the mbtiles. 
![image](https://user-images.githubusercontent.com/103890980/230871962-85308543-3c44-4eed-9025-ff40987aaa2c.png)

**Reply by satya-auti** [reply]
Is there any other solution for this? @systemed

**Reply by systemed** [reply]
Can you upload the pbf somewhere?

**Reply by satya-auti** [reply]
Wait will share a link of that pbf file.

**Reply by satya-auti** [reply]
https://www.dropbox.com/s/if15chvvgzwun8y/hyderabad_planet_v0_6_2.osm.pbf?dl=0

https://www.dropbox.com/s/eqzz4dms7z56hkt/hyderabad_planet_v0_6_1.osm.pbf?dl=0

@systemed 
I have shared both links with you try anyone of that. Both generated using osmium ` --strategy=smart`.

**Reply by satya-auti** [reply]
@systemed  you can try with anyone of above link.

**Reply by systemed** [reply]
````
osmium fileinfo -e ~/Downloads/hyderabad_planet_v0_6_2.osm.pbf
  Largest node ID: 5000320530230
  Largest way ID: 3000024385671
  Largest relation ID: 3000000000064
````

You have objects with very large IDs in your data. tilemaker is written for use with OSM data and expects IDs to be within the range of OSM data. If you want to use this file (presumably it has some third-party data?) you'll need to use `osmium renumber` to get it within the normal ID space of OSM data:

````
$ osmium renumber ~/Downloads/hyderabad_planet_v0_6_2.osm.pbf -o ~/Downloads/hyderabad_renumbered.osm.pbf
[======================================================================] 100% 
$ tilemaker --input ~/Downloads/hyderabad_renumbered.osm.pbf --output ~/Downloads/hyderabad.mbtiles --bbox 72.9719,17.8178,75.0314,19.3831
````

**Reply by satya-auti** [reply]
Means after renumbering it will work?

**Reply by systemed** [reply]
Yes.

**Reply by satya-auti** [reply]
Thank You.....!!! @systemed .It's working fine after renumbering OSM ID's.

**Reply by systemed** [reply]
No problem!
